### PR TITLE
[WIP] Bug 1918392: UpdateHAProxy

### PR DESCRIPTION
**Pull request by candita**
Test the change to HAProxy version 1.8.28 needed by BZ 1918392

### CODE REVIEW

File: images/router/haproxy/Dockerfile [hunk 1]
```diff
@@ -5,11 +5,7 @@
...
 FROM openshift/origin-cli
 
-COPY rhel-server-ose.repo /etc/yum.repos.d
-# If this is a non-CI/RHEL build, the repo is not necessary
-RUN test -e /etc/centos-release || rm /etc/yum.repos.d/rhel-server-ose.repo
-
-RUN INSTALL_PKGS="haproxy18 rsyslog" && \
+RUN INSTALL_PKGS="https://raw.githubusercontent.com/candita/haproxy-testing/main/haproxy18-1.8.28-1.el7.x86_64.rpm rsyslog" && \
     yum install -y $INSTALL_PKGS && \
     rpm -V $INSTALL_PKGS && \
```
Comment (Miciah) [inline images/router/haproxy/Dockerfile:10]:
```suggestion
RUN INSTALL_PKGS="rsyslog" && \
    yum install -y https://raw.githubusercontent.com/candita/haproxy-testing/main/haproxy18-1.8.28-1.el7.x86_64.rpm $INSTALL_PKGS && \
    rpm -V haproxy18 $INSTALL_PKGS && \
```

**Comment by candita** [comment]
W0204 16:09:03.941979   11441 dockerimagelookup.go:233] Docker registry lookup failed: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
error: unable to locate any images in image streams, local docker images with name "centos/ruby-25-centos7"

/test cmd

**Comment by candita** [comment]
W0204 18:08:03.666316   11647 dockerimagelookup.go:233] Docker registry lookup failed: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
error: unable to locate any images in image streams, local docker images with name "centos/ruby-25-centos7"
/retest

**Comment by candita** [comment]
W0204 19:26:28.963003   11103 dockerimagelookup.go:233] Docker registry lookup failed: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
error: no tags found on matching image stream: "openshift/nodejs"
/retest

**Comment by candita** [comment]
W0204 20:28:13.131001   11675 dockerimagelookup.go:233] Docker registry lookup failed: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
error: unable to locate any images in image streams, local docker images with name "centos/ruby-25-centos7"
/retest

**Comment by candita** [comment]
0204 21:28:46.758589   10941 dockerimagelookup.go:233] Docker registry lookup failed: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
error: no tags found on matching image stream: "openshift/nodejs"
/retest

File: images/router/haproxy/Dockerfile [hunk 2]
```diff
@@ -5,13 +5,12 @@
...
-# If this is a non-CI/RHEL build, the repo is not necessary
-RUN test -e /etc/centos-release || rm /etc/yum.repos.d/rhel-server-ose.repo
+RUN yum install -y https://raw.githubusercontent.com/candita/haproxy-testing/main/haproxy18-1.8.28-1.el7.x86_64.rpm 
+RUN haproxy -vv
 
-RUN INSTALL_PKGS="haproxy18 rsyslog" && \
+RUN INSTALL_PKGS="rsyslog" && \
     yum install -y $INSTALL_PKGS && \
-    rpm -V $INSTALL_PKGS && \
+    rpm -V haproxy18 $INSTALL_PKGS && \
```
Comment (frobware) [inline images/router/haproxy/Dockerfile:13]: Drop the reference to `haproxy18` it should have been installed by the `yum install`. At the very least just install the one we want to test CI with.

Comment (frobware) [reply images/router/haproxy/Dockerfile:13]: Or move the `RUN haxproxy -vv` to after all the package installs to verify what version we're testing with post RPM install.

Comment (candita) [reply images/router/haproxy/Dockerfile:13]: That works for `yum` but not for `rpm`.  I had to use the alias for `rpm -V`

Comment (frobware) [reply images/router/haproxy/Dockerfile:13]: OK, I misread this a bit. Apologies. The times I have done this I avoid any reference to `haproxy18` as I install it directly from a known place and the `haproxy -vv` will confirm build+version info. This is fine. Apologies for the noise; it just looked different to the patterns I had ...

Comment (candita) [reply images/router/haproxy/Dockerfile]: You have a point, it's better to keep it as clean as possible for future reference, so I updated it.

**Comment by candita** [comment]
W0209 20:18:26.740538   11715 dockerimagelookup.go:233] Docker registry lookup failed: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
/test cmd

**Comment by candita** [comment]
++ Building go targets for linux/amd64: tools/etcdhelper
[INFO] hack/build-go.sh exited with code 0 after 00h 00m 59s
ERROR: unable to create client config: open /tmp/openshift/clusterup/openshift.local.config/master/master.etcd-client.crt: no such file or directory
/retest

**Comment by candita** [comment]
Running test/extended/clusterup.sh:63: executing 'oc get is -n openshift ruby -o jsonpath='{ .status.tags[*].tag }'' expecting any result and text 'latest'; re-trying every 1s until completion or 600.000s...
FAILURE after 600.275s: test/extended/clusterup.sh:63: executing 'oc get is -n openshift ruby -o jsonpath='{ .status.tags[*].tag }'' expecting any result and text 'latest'; re-trying every 1s until completion or 600.000s: the command timed out
Standard output from the command:
2.0 2.2 2.3 2.4 2.5
... repeated 488 times
Standard error from the command:
[ERROR] PID 1350: hack/lib/cmd.sh:617: `return "${return_code}"` exited with status 1.

/retest

**Comment by candita** [comment]
AILURE after 20.027s: test/extended/clusterup.sh:301: executing 'timeout 20s oc run peer --image=openshift/origin-gitserver:latest --attach --restart=Never --command -- curl http://git:8080/_/healthz' expecting success: the command returned the wrong error code
There was no output from the command.
Standard error from the command:
If you don't see a command prompt, try pressing enter.

/retest

**Comment by candita** [comment]
W0210 16:46:29.044813   17543 dockerimagelookup.go:233] Docker registry lookup failed: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
error: unable to locate any images in image streams, local docker images with name "centos/ruby-25-centos7"

/retest

**Comment by candita** [comment]
W0210 17:43:50.933810   17687 dockerimagelookup.go:233] Docker registry lookup failed: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
error: unable to locate any images in image streams, local docker images with name "centos/ruby-25-centos7"

/retest

**Comment by candita** [comment]
You have reached your pull rate limit.
/retest

**Comment by candita** [comment]
FAILURE after 600.238s: test/extended/clusterup.sh:63: executing 'oc get is -n openshift ruby -o jsonpath='{ .status.tags[*].tag }'' expecting any result and text 'latest'; re-trying every 1s until completion or 600.000s: the command timed out
Standard output from the command:
2.0 2.2 2.3 2.4 2.5
... repeated 486 times

/retest

**Comment by candita** [comment]
FAILURE after 599.645s: test/extended/clusterup.sh:63: executing 'oc get is -n openshift ruby -o jsonpath='{ .status.tags[*].tag }'' expecting any result and text 'latest'; re-trying every 1s until completion or 600.000s: the command timed out
Standard output from the command:
2.0 2.2 2.3 2.4 2.5
... repeated 487 times

/retest

**Comment by candita** [comment]
Running test/extended/clusterup.sh:301: executing 'timeout 20s oc run peer --image=openshift/origin-gitserver:latest --attach --restart=Never --command -- curl http://git:8080/_/healthz' expecting success...
FAILURE after 20.027s: test/extended/clusterup.sh:301: executing 'timeout 20s oc run peer --image=openshift/origin-gitserver:latest --attach --restart=Never --command -- curl http://git:8080/_/healthz' expecting success: the command returned the wrong error code
There was no output from the command.

/retest

**Comment by candita** [comment]
FAILURE after 600.801s: test/extended/clusterup.sh:63: executing 'oc get is -n openshift ruby -o jsonpath='{ .status.tags[*].tag }'' expecting any result and text 'latest'; re-trying every 1s until completion or 600.000s: the command timed out
Standard output from the command:
2.0 2.2 2.3 2.4 2.5
... repeated 480 times

/test extended_clusterup

**Comment by candita** [comment]
W0211 16:37:10.212628   17243 dockerimagelookup.go:233] Docker registry lookup failed: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit

/test extended_clusterup

**Comment by candita** [comment]
FAILURE after 599.689s: test/extended/clusterup.sh:63: executing 'oc get is -n openshift ruby -o jsonpath='{ .status.tags[*].tag }'' expecting any result and text 'latest'; re-trying every 1s until completion or 600.000s: the command timed out
Standard output from the command:
2.0 2.2 2.3 2.4 2.5
/retest

**Comment by candita** [comment]
W0212 15:39:05.124959   17101 dockerimagelookup.go:233] Docker registry lookup failed: toomanyrequests: You have reached your pull rate limit. 
/retest

**Comment by candita** [comment]
Running test/extended/clusterup.sh:301: executing 'timeout 20s oc run peer --image=openshift/origin-gitserver:latest --attach --restart=Never --command -- curl http://git:8080/_/healthz' expecting success...
FAILURE after 20.029s: test/extended/clusterup.sh:301: executing 'timeout 20s oc run peer --image=openshift/origin-gitserver:latest --attach --restart=Never --command -- curl http://git:8080/_/healthz' expecting success: the command returned the wrong error code
There was no output from the command.

/retest

**Comment by candita** [comment]
Tests completed successfully.  Image is loaded to mirror.  This PR is no longer needed for testing purposes.

**Pull request closed without merging**